runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptionsTest.java

```diff
@@ -90,4 +93,40 @@ public void testUtf8UserNameAndApplicationNameIsNormalized() {
     options.setAppName("fəˈnɛtık əsoʊsiˈeıʃn");
     assertEquals("f00n0t0k00so0si0e00n-0i00nt00n000n0l0-1208190706", options.getJobName());
   }
+
+  @Test
```

Comment:
please do much more comprehensive testing here, including of `gcpTempLocation` and all reasonable error cases.